src/clj_kondo/impl/analyzer.clj

```diff
@@ -240,15 +240,27 @@
                         (str "unsupported binding form " expr))))
          :vector (let [children (:children expr)
                        all-tokens? (every? #(identical? :token %) (map :tag children))
+                       as-binding? (identical? :as (some-> (butlast children)
+                                                           last
+                                                           :k
+                                                           name
```

Comment:
I haven't tried locally yet, but does this call `name` on a rewrite-clj node? That seems a bit weird to me.

Is there a more efficient way to get to the but-last element of a seq without walking it almost twice? Maybe not, but maybe we could prevent doing that if we know the linter isn't enabled (which is checked after this, so should be moved up)

Reply:
I borrowed code from another part of the app. As far as I can tell, calling `(keyword (name (:k some-node)))` is unnecessary word. For keywords, the value of `:k` will be a keyword, and for all else, it will be `nil`.

You're right that I can do this slightly more efficiently. I've pushed a new version that first checks if the flag is sent, and if so, uses `drop` to get the last two items, and if the second-to-last item is `:as`, return the last item `as-sym`.

Thanks for the idea, glad to get it simpler and faster.